pkg/gateway.go

```diff
@@ -15,15 +14,15 @@ import (
 // It loads configs, start API server(s), and listen to termination signals to shut down
 type Gateway struct {
 	// serverManager controls API over different protocols
-	serverManager *api.ServerManager
+	serverManager *http.ServerManager
 	// signalChannel is used to receive termination signals from the OS.
 	signalC chan os.Signal
 	// shutdownC is used to terminate the gateway
 	shutdownC chan struct{}
 }
 
 func NewGateway() (*Gateway, error) {
-	serverManager, err := api.NewServerManager(&http.HTTPServerConfig{})
+	serverManager, err := http.NewServerManager(&http.HTTPServerConfig{})
```

Comment:
ServerManager works with all API types Glide will support. `pkg/api` is the right place for it. It's not specific to http service (that is only supported right now)